ddtrace/internal/dm.py

```diff
@@ -0,0 +1,100 @@
+import enum
+from hashlib import sha256
+import re
+from typing import TYPE_CHECKING
+
+from ddtrace import config
+from ddtrace.internal.constants import DEFAULT_SERVICE_NAME
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+if TYPE_CHECKING:
+    from typing import Optional
+    from typing import Text
+    from typing import Union
+
+    from ddtrace.context import Context
+
+
+class SamplingMechanism(enum.Enum):
+    UNKNOWN = -1
+    DEFAULT = 0
+    AGENT_RATE = 1
+    REMOTE_RATE = 2
+    TRACE_SAMPLING_RULE = 3
+    MANUAL = 4
+    APPSEC = 5
+    REMOTE_RATE_USER = 6
+    REMOTE_RATE_DATADOG = 7
+
+
+SERVICE_HASH_MAX_LENGTH = 10
+TRACE_TAG_KEY = "_dd.p.dm"
+
+# Use regex to validate trace tag value
+_RE_TRACE_TAG = re.compile(r"^([0-9a-f]{0,10})-([0-8])$")
```

Comment:
I would update this to be `-([0-9]+)` to support any future sampling mechanism values

We also have `-1`, do we need to support that here as well?

Reply:
This should also be `0` or `10` hex characters, not 0 to 10

Reply:
```suggestion
_RE_TRACE_TAG = re.compile(r"^([0-9a-f]{10}?)-([0-8])$")
```

(not sure my suggestion is correct, regex is hard)